.circleci/config.yml

```diff
@@ -32,17 +32,19 @@ jobs:
             sudo apt-get update -y;
             sudo apt-get install -y g++-4.8 curl libdw-dev libelf-dev elfutils binutils-dev libssl-dev pkg-config zlib1g-dev
       - restore_cache:
-          name: Restoring ~/.local Cache
+          # Actually master does not need to download the cache. However we cannot make conditions here.
+          name: Downloading ~/.cargo and ~/.local Cache from master
           keys:
-            - v2-local-{{ .Branch }}-{{ .Environment.CIRCLE_SHA1 }}
-            - v2-local-{{ .Branch }}
-            - v2-local
-      - restore_cache:
-          name: Restoring ~/.cargo Cache
-          keys:
-            - v2-cargo-{{ .Branch }}-{{ checksum "Cargo.lock" }}
-            - v2-cargo-{{ .Branch }}
-            - v2-cargo
+            - v3-cargo-master
```

Comment:
Will it choose the most recent cache that starts with `v3-cargo-master`?

Reply:
Because PR's build will not upload valid cache (see readme idea 2). So for individual PRs, they receive snapshot from latest master, hoping for some usefulness.
Master does not need to download cache, but it is not possible to write conditional statement here, so leave it.

Reply:
Why not specify master's git hash? 

```
git rev-parse master
```

Reply:
First, you can't write anything dynamic here, so using a non const value is nearly impossible. (Notice that environment variable is also useless because in `restore_cache`, it only supports built-in environment variables and context variables).

Second, CircleCI matches cache by prefix and if there are multiple matches a latest one is used. So this is enough for us.

Reply:
I found out that I misunderstood your comment yesterday LOL the comment above should be the right answer. Yes, CircleCI will match to the most recent cache in that prefix, it is a documented behaviour.